### PR TITLE
Further fixes for #1781 (preference search exceptions)

### PR DESCRIPTION
**Pull request by Lupum1001**
Added some more missing Preference keys to fix the behavior in preference searches. Fixed some whitespace and attributes along the way.

### CODE REVIEW

**Comment by tolot27** [comment]
> Is there a way to screen for this issue automatically, either by linting the xml or writing unit tests?

Not really automatically, but you can use XPath to query nodes which does not have a specific attribute. You can test it with http://xpather.com and load `pref_data_sync.xml`. The xpath query would be `//Preference[not(@android:key)]` or `//PreferenceScreen[not(@android:key)]`.

Initially, I thought the lint check `MissingDefaultResource` will do the job but it does not.

**Comment by tolot27** [comment]
> `//PreferenceScreen[not(@android:key)]`

This will also find root nodes of `PreferenceScreen`. I don't know whether they need an `android:key` attribute or not.

File: app/src/main/res/xml/xdrip_plus_prefs.xml [hunk 1]
```diff
@@ -1198,7 +1181,6 @@
                 android:maxLength="4"
                 android:title="@string/title_pendiq_pin" />
 
-
         </PreferenceScreen>
         </PreferenceScreen>
```
Comment (tolot27) [inline app/src/main/res/xml/xdrip_plus_prefs.xml:1185]: Here is still an indentation missing. Did you use code formatting?

Comment (Lupum1001) [reply app/src/main/res/xml/xdrip_plus_prefs.xml:1185]: I didn't, no. I just manually corrected some of the more obvious ones. There's enough inconsistency in the formatting that the whitespace changes would have gotten out of hand really fast. As long as it was readable and more or less consistent with its surroundings, I tended not to mess with it.

**Pull request merged**